src/Evolution/Systems/GeneralizedHarmonic/Actions/NumericInitialData.hpp

```diff
@@ -0,0 +1,325 @@
+      db::DataBox<DbTagsList>& box,
+      const tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+      Parallel::GlobalCache<Metavariables>& cache,
+      const ArrayIndex& /*array_index*/, const ActionList /*meta*/,
+      const ParallelComponent* const /*meta*/) {
+    // Select the subset of the available variables that we want to read from
+    // the volume data file
+    const auto& selected_initial_data_vars =
+        get<detail::Tags::NumericInitialDataVariables<ImporterOptionsGroup>>(
+            box);
+    tuples::tagged_tuple_from_typelist<
+        db::wrap_tags_in<importers::Tags::Selected, detail::all_numeric_vars>>
+        selected_fields{};
+    std::visit(
+        [&selected_fields](const auto& vars) {
+          // This lambda is invoked with the set of vars selected in the input
+          // file, which map to the tensor names that should be read from the H5
+          // file
+          using selected_vars = std::decay_t<decltype(vars)>;
+          // Get the mapped tensor name from the input file and select it in the
+          // set of all possible vars.
+          tmpl::for_each<typename selected_vars::tags_list>(
+              [&selected_fields, &vars](const auto tag_v) {
+                using tag = typename std::decay_t<decltype(tag_v)>::type::tag;
+                get<importers::Tags::Selected<tag>>(selected_fields) =
+                    get<detail::OptionTags::VarName<tag>>(vars);
+              });
+        },
+        selected_initial_data_vars);
+    // Dispatch loading the variables from the volume data file
+    // - Not using `ckLocalBranch` here to make sure the simple action
+    //   invocation is asynchronous.
+    auto& reader_component = Parallel::get_parallel_component<
+        importers::ElementDataReader<Metavariables>>(cache);
+    Parallel::simple_action<importers::Actions::ReadAllVolumeDataAndDistribute<
+        ImporterOptionsGroup, detail::all_numeric_vars, ParallelComponent>>(
+        reader_component, std::move(selected_fields));
+    return {std::move(box)};
+  }
+};
+
+/*!
+ * \brief Receive numeric initial data loaded by
+ * GeneralizedHarmonic::Actions::ReadNumericInitialData.
+ *
+ * Place this action in the action list after
+ * GeneralizedHarmonic::Actions::ReadNumericInitialData to wait until the data
+ * for this element has arrived, and then transform the data to GH variables and
+ * store it in the DataBox to be used as initial data.
+ *
+ * This action modifies the following tags in the DataBox:
+ * - gr::Tags::SpacetimeMetric<3, Frame::Inertial, DataVector>
+ * - GeneralizedHarmonic::Tags::Pi<3, Frame::Inertial>
+ * - GeneralizedHarmonic::Tags::Phi<3, Frame::Inertial>
+ * - ::Tags::Time: To compute numerical derivatives on time-dependent domains we
+ *   have to temporarily set the time to the initial time, so it gets picked up
+ *   by compute items for the Jacobian. We set the time back to NaN before this
+ *   action completes.
```

Comment:
@wthrowe I had to put in this workaround for an issue that @geoffrey4444 encountered with time-dependent domains. Essentially, `::Tags::Time` is NaN during initialization, but for taking numeric derivatives of the initial data we have to set it to the `InitialTime` temporarily. When you get the chance, could we try to come up with a more sustainable solution? Happy to discuss at in a call some time.

Reply:
Just set it where it's first created.

Reply:
So you mean, set `::Tags::Time` to `InitialTime` in `Initialization::Actions::TimeAndTimeStep`?

Reply:
Use `initial_time.value()` rather than the input parameter, but otherwise yes.  (They should be the same, but this is consistent with what is done elsewhere.)